src/log-entry-dashboard/log-entry-form/log-entry-form.component.tsx

```diff
@@ -20,124 +20,305 @@ import {
-import { inputClass } from './log-entry-form.styles';
-import { panelClass } from '../log-entry-dashboard.styles';
+import { timeForm, buttonContainer } from './log-entry-form.styles';
+import { RadioChangeEvent } from 'antd/lib/radio';
+import { any } from 'prop-types';
 
 const FormItem = Form.Item;
+const RadioButton = Radio.Button;
+const RadioGroup = Radio.Group;
+const { TextArea } = Input;
+const SelectOption = Select.Option;
 
 interface LogEntryFormProps extends FormComponentProps {
   authContext: IAuthenticationContext;
 }
 
 class LogEntryFormComponent extends React.Component<LogEntryFormProps> {
+  state = {
+    categoryId: 0,
+    pointsCompeleted: 'notCompleted',
+  };
+
+  handleHourChange = (e?: number) => {
+    const {
+      form: { setFieldsValue },
+    } = this.props;
+
+    setFieldsValue({ logValue: 0 });
+  };
+
+  handleCategoryChange = (e: string) => {
+    this.setState({ categoryId: e });
+  };
+
+  handleRaidoChange = (e: RadioChangeEvent) => {
+    if (e.target.value === 'completed') {
+      this.setState({ pointsCompeleted: 'completed' });
+    } else if (e.target.value === 'notCompleted') {
+      this.setState({ pointsCompeleted: 'notCompleted' });
+    }
+  };
+
+  hasErrors = (fieldsError: any) => {
+    return Object.keys(fieldsError).some(field => fieldsError[field]);
+  };
+
   handleOnSubmit = (e: React.SyntheticEvent) => {
     e.preventDefault();
     const {
       authContext: { currentUser },
       form,
     } = this.props;
+
     form.validateFields((err, values) => {
       if (err) {
         return;
       }
 
       createLogEntry({
-        title: values['description'],
-        arePointsCompleted: values['completedPoints'],
+        title: values['title'],
+        arePointsCompleted: values['completedPoints'] || null,
         notes: values['notes'],
         date: values['logDate'],
         value: values['logValue'],
         categoryId: values['category'], //Categories.BillableTime,
-        subCategoryId: values['subcategory'], //SubCategories.Admin,
+        subCategoryId: values['subcategory'] || null, //SubCategories.Admin,
         updatedBy: currentUser ? currentUser.id : '',
         createdAt: new Date(),
         updatedAt: new Date(),
       });
+
+      form.resetFields();
     });
   };
 
-  render() {
+  handleClearForm = () => {
+    const { form } = this.props;
+    form.resetFields();
+  };
+
+  renderCategoriesSelectOptions = () => {
+    let keys = Object.keys(Categories).filter(
+      (key: any) => !isNaN(Number(Categories[key]))
+    );
+
+    return keys.map((category: any) => {
+      return (
+        <SelectOption key={category} value={Categories[category]}>
+          {category}
+        </SelectOption>
+      );
+    });
+  };
+
+  renderSubCategoriesSelectOptions = () => {
+    let keys = Object.keys(SubCategories).filter(
+      (key: any) => !isNaN(Number(SubCategories[key]))
+    );
+
+    return keys.map((subCategory: any) => {
```

Comment:
`any`

Reply:
have not found a type that works here other than 'any'. I even tried using | between multiple values.